src/sentry/integrations/vercel/uninstall.py

```diff
@@ -1,17 +1,104 @@
+import hashlib
+import hmac
 import logging
 
+from django.utils.crypto import constant_time_compare
 from django.views.decorators.csrf import csrf_exempt
+from requests.exceptions import RequestException
 
+from sentry import VERSION, http, options
 from sentry.api.base import Endpoint
-from sentry.models import AuditLogEntryEvent, Integration, Organization, OrganizationIntegration
+from sentry.models import (
+    AuditLogEntryEvent,
+    Integration,
+    Organization,
+    OrganizationIntegration,
+    Project,
+    SentryAppInstallationForProvider,
+    SentryAppInstallationToken,
+)
+from sentry.shared_integrations.exceptions import IntegrationError
 from sentry.utils.audit import create_audit_entry
+from sentry.utils.compat import filter
+from sentry.utils.http import absolute_uri
 
 logger = logging.getLogger("sentry.integrations.vercel.uninstall")
 
 
+class NoCommitFoundError(IntegrationError):
+    pass
+
+
+def verify_signature(request):
+    signature = request.META["HTTP_X_VERCEL_SIGNATURE"]
+    secret = options.get("vercel.client-secret")
+
+    expected = hmac.new(
+        key=secret.encode("utf-8"), msg=bytes(request.body), digestmod=hashlib.sha1
+    ).hexdigest()
+    return constant_time_compare(expected, signature)
+
+
+def safe_json_parse(resp):
+    if resp.headers.get("content-type") == "application/json":
+        return resp.json()
+    return None
+
+
+def get_payload_and_token(payload, organization_id, sentry_project_id):
+    meta = payload["deployment"]["meta"]
+
+    # look up the project so we can get the slug
+    project = Project.objects.get(id=sentry_project_id)
+
+    # find the connected sentry app installation
+    installation_for_provider = SentryAppInstallationForProvider.objects.select_related(
+        "sentry_app_installation"
+    ).get(organization_id=organization_id, provider="vercel")
+    sentry_app_installation = installation_for_provider.sentry_app_installation
+
+    # find a token associated with the installation so we can use it for authentication
+    sentry_app_installation_token = (
+        SentryAppInstallationToken.objects.select_related("api_token")
+        .filter(sentry_app_installation=sentry_app_installation)
+        .first()
+    )
+    if not sentry_app_installation_token:
+        raise SentryAppInstallationToken.DoesNotExist()
+
+    # find the commmit sha so we can  use it as as the release
+    commit_sha = (
+        meta.get("githubCommitSha") or meta.get("gitlabCommitSha") or meta.get("bitbucketCommitSha")
```

Comment:
@MeredithAnya I believe Vercel has added something like `gitCommitSha` so you don't have to OR them like this

Reply:
@scefali couldn't find any documentation for this, and when I tested it out, the old payload and new payload are the same so I'm going to keep this.